lib/meta-data-helpers.js

```diff
@@ -0,0 +1,32 @@
+const buildMetaDataObj = ({
+  resource = {},
+  md5,
+  localInfo = {},
+  remoteInfo = {}
+}) => {
```

Comment:
I *think* if you are handed a `remoteInfo`, you do not need `md5` cause `remoteInfo` will have it

also, I think you should make it obvious possibly with some checks in here what combos of params you expect/allow - i.e., you would never get all of these handed to you at once.